rfcs/0029-backports-team.md

```diff
@@ -0,0 +1,256 @@
+---
+feature: backports_team
+start-date: 2018-07-28
+author: samueldr
+co-authors: ...
+related-issues: ...
+---
+
+Summary
+=======
+[summary]: #summary
+
+This intends to formalize the backports process and formalize the formation of
+a team managing backports.
+
+
+Motivation
+==========
+[motivation]: #motivation
+
+The way it is currently handled makes it so too often fixes for lesser-used
+software are not backported to the *current-stable* revision of NixOS.
+Furthermore, even some more-often used software sometimes lag behind since the
+updates were simply not cherry-picked. While there are [general
+guidelines][grahamc_backport] written up, no canonical process is documented.
+
+Backports is not only a security feature, but also a user-experience feature.
+Upstream software is getting updated; it would be best if current-stable does
+not get the reputation of being stale.
+
+By formalizing the process, it is expected to create a good hygiene that will
+keep NixOS current-stable fresher, instead of presenting stale software to the
+end-users.
+
+Finally, by appointing a team, even if fluid, it is expected that the
+responsibilities given to the users will _nudge them_ toward doing the work
+required of maintaining the backports. The work of integrating backports should
+be the release manager's, as per [RFC 0015][rfc0015]. Through [dubious
+stats][stats_release], it seems that for the 18.03 release their work has
+been minimal, and instead filled-in by ad-hoc work by other members.
+
+
+Detailed design
+===============
+[design]: #detailed-design
+
+Backports process
+-----------------
+
+Expected to be backported to current-stable are the following:
+<sup>[1][grahamc_backport]</sup>
+
+ * Patch-level and minor-level updates.
+ * Security patches from downstream projects (other distributions) when
+   no patch-level or minor-level update is made available.
+ * Any update when the current stable version is completely broken.
+ * Extremely security-sensitive software, in particular web browsers,
+   mail user-agents and the kernel, can and should be updated to the version
+   expected by upstream to be secure.
+
+### Examples
+
+An example of broken software needing a major update to be backported is
+Spotify, which regularly breaks with old versions.
```

Comment:
I'm hoping to fix this with https://github.com/NixOS/nixpkgs/pull/44123.

Reply:
Awesome, said example was gleaned from the [general guidelines](https://gist.github.com/grahamc/c60578c6e6928043d29a427361634df6#what-to-backport) linked earlier. I'm sure as a group we can either abstract away the exact software which break as any one of those, or think up a small list of "glassware" (often-breaking software).

Reply:
Anything that can't be hosted on nixos.org is a candidate for this. `grep` can find more than a couple. (It also finds things like tor-browser-bundle, which we probably could rehost but haven't, and which is a prime candidate for backporting.)